src/plugins/intel_gpu/src/plugin/transformations/broadcast_reshape_matmul_fusion.cpp

```diff
@@ -0,0 +1,145 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "broadcast_reshape_matmul_fusion.hpp"
+
+#include "intel_gpu/op/gemm.hpp"
+
+#include "openvino/core/rt_info.hpp"
+#include "openvino/op/broadcast.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/reshape.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "openvino/pass/pattern/op/or.hpp"
+#include "transformations/utils/utils.hpp"
+
+namespace ov {
+namespace intel_gpu {
+
+BroadcastReshapeMatmulFusion::BroadcastReshapeMatmulFusion() {
+    using namespace ov::pass::pattern;
+
+    auto not_reshape = [](const ov::Output<ov::Node>& output) -> bool {
+        return std::dynamic_pointer_cast<ov::op::v1::Reshape>(output.get_node_shared_ptr()) == nullptr;
+    };
+
+    auto broadcast_rank_equals_and_has_static_dims = [](const ov::Output<ov::Node>& output) -> bool {
+        return rank_equals(5)(output) && has_static_dims({2, 3}) && consumers_count(1);
+    };
+
+    auto reshape_rank_equals_and_has_static_dim = [](const ov::Output<ov::Node>& output) -> bool {
+        return rank_equals(4)(output) && has_static_dim(2) && consumers_count(1);
+    };
+
+    auto input_a_m = any_input(not_reshape);
+    auto input_b_m = any_input(not_reshape);
+
+    auto broadcast_a_target_shape_m = wrap_type<ov::op::v0::Constant>();
+    auto broadcast_a_m = wrap_type<ov::op::v3::Broadcast>({input_a_m, broadcast_a_target_shape_m}, broadcast_rank_equals_and_has_static_dims);
+    auto broadcast_b_target_shape_m = wrap_type<ov::op::v0::Constant>();
+    auto broadcast_b_m = wrap_type<ov::op::v3::Broadcast>({input_b_m, broadcast_b_target_shape_m}, broadcast_rank_equals_and_has_static_dims);
```

Comment:
Why target shape is expected to be constant? It limits pattern applicability as target shape could be computed in shape of subgraph

Reply:
Expanding the pattern for broadcast/reshape with non constant target shape/output pattern will proceed to the next task and follow up with a separate PR